pkg/wasm/watcher.go

```diff
@@ -104,6 +104,26 @@ func addWatchFile(cfg *filterConfigItem, factory *FilterConfigFactory) {
 	log.DefaultLogger.Infof("[proxywasm] [watcher] addWatchFile start to watch wasm file and its dir: %s", path)
 }
 
+// remove watching file
+func removeWatchFile(cfg *filterConfigItem) {
+	path := cfg.VmConfig.Path
+	// Add starts watching the named file or directory (non-recursively).
+	if err := watcher.Remove(path); err != nil {
+		log.DefaultLogger.Errorf("[proxywasm] [watcher] removeWatchFile fail to stop watch wasm file, err: %v", err)
+	}
+
+	dir := filepath.Dir(path)
+	if err := watcher.Remove(dir); err != nil {
+		log.DefaultLogger.Errorf("[proxywasm] [watcher] removeWatchFile fail to stop watch wasm dir, err: %v", err)
+		return
+	}
```

Comment:
Here you also need to determine if there are other files being watched in the same directory